# machine-learning-resource

### courses, tools and reviews

* Big data and deep learning for RNA biology https://www.nature.com/articles/s12276-024-01243-w
* google machine learning free classes: https://developers.google.com/machine-learning
* https://vas3k.com/blog/machine_learning/
* statquest https://app.learney.me/maps/StatQuest
* [Scientific discovery in the age of artificial intelligence](https://www.nature.com/articles/s41586-023-06221-2)
* [Avoiding common pitfalls in machine learning omic data science](https://www.nature.com/articles/s41563-018-0241-z). see PDF in the repo.
* [Incorporating Machine Learning into Established Bioinformatics Frameworks](https://www.cell.com/immunity/fulltext/S1074-7613(23)00129-2?)
* [Ten quick tips for deep learning in biology](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8946751/)
* [How to avoid machine learning pitfalls:a guide for academic researchers](https://arxiv.org/abs/2108.02497)
* [Artificial intelligence for multimodal data integration in oncology](https://www.cell.com/cancer-cell/fulltext/S1535-6108(22)00441-X)
* [Steps to avoid overuse and misuse of machine learning in clinical research](https://www.nature.com/articles/s41591-022-01961-6)
* [Navigating the pitfalls of applying machine learning in genomics](https://www.nature.com/articles/s41576-021-00434-9). Nature Review Genetics.
* [A guide to machine learning for biologists](https://www.nature.com/articles/s41580-021-00407-0)
* [Deep learning for computational biology](https://www.embopress.org/doi/full/10.15252/msb.20156651)
* [Current progress and open challenges for applying deep learning across the biosciences](https://www.nature.com/articles/s41467-022-29268-7)
* [paper: A pitfall for machine learning methods aiming to predict across cell types](https://www.biorxiv.org/content/early/2019/01/04/512434)
* [Machine Learning for Integrating Data in Biology and Medicine: Principles, Practice, and Opportunities](https://arxiv.org/abs/1807.00123)
* [Neural Networks and Deep Learning](http://neuralnetworksanddeeplearning.com/) A free online book by Michael Nielsen.
* [supervised machine learning case study in R](https://supervised-ml-course.netlify.com/) A Free, Interactive Course Using Tidy Tools
* [Machine Learning for Everyone](https://vas3k.com/blog/machine_learning/)
* [An Introduction to Statistical Learning](https://www-bcf.usc.edu/~gareth/ISL/)
* [Hands-on Machine Learning with R](https://bradleyboehmke.github.io/HOML/)
* [Practical Deep Learning For Coders from fast ai](http://course.fast.ai/) 
* [r2d3: A visual introduction to machine learning](http://www.r2d3.us/)
* [parsnip](https://github.com/tidymodels/parsnip)A tidy unified interface to models https://tidymodels.github.io/parsnip/
* Descriptive mAchine Learning EXplanations: [DALEX](https://pbiecek.github.io/DALEX/)
* [machine learning](https://ocw.mit.edu/courses/electrical-engineering-and-computer-science/6-867-machine-learning-fall-2006/) MIT OCW
* [CS229: Machine Learning - The Summer Edition](https://cs229.stanford.edu/syllabus-summer2020.html) free course from Stanford.

### deep learning

* [nice blog posts on deep NN](https://colah.github.io/)
* [Yann LeCun’s Deep Learning Course at CDS](https://cds.nyu.edu/deep-learning/)
* [Dive into Deep Learning](https://d2l.ai/) A free interactive book. really nice!
* [neuralnetworks and deeplearning](http://neuralnetworksanddeeplearning.com/) A free ebook
* http://introtodeeplearning.com/ MIT course
* [Opportunities and obstacles for deep learning in biology and medicine: 2019 update](https://greenelab.github.io/deep-review/)
* [Machine Learning for Beginners: An Introduction to Neural Networks](https://victorzhou.com/blog/intro-to-neural-networks/) A blog post. see how linear algebra is playing here!
* [Practical Deep Learning for Coders, 2019 edition](https://course.fast.ai/) course by fast.ai
* [Examples of using deep learning in Bioinformatics](https://github.com/lykaust15/Deep_learning_examples)
* [Getting started with deep learning in R](https://blogs.rstudio.com/tensorflow/posts/2018-09-07-getting-started/) blog post from Rstudio.  
* [deep learning biology](https://github.com/hussius/deeplearning-biology)
* [Bayesian deep learning for single-cell analysis](https://www.nature.com/articles/s41592-018-0230-9)
* [A primer on deep learning in genomics](https://www.nature.com/articles/s41588-018-0295-5)
* [Selene: a PyTorch-based deep learning library for biological sequence-level data](https://www.biorxiv.org/content/early/2018/12/14/438291)
* [Janggu - Deep learning for Genomics](https://github.com/BIMSBbioinfo/janggu)
* [Dive into Deep Learning](http://en.diveintodeeplearning.org/) An interactive deep learning book for students, engineers, and researchers.
* [Using Nucleus and TensorFlow for DNA Sequencing Error Correction](https://medium.com/tensorflow/using-nucleus-and-tensorflow-for-dna-sequencing-error-correction-47f3f7fc1a50) [tutorial](https://colab.research.google.com/github/google/nucleus/blob/master/nucleus/examples/dna_sequencing_error_correction.ipynb)
* [SingleCellNet: a computational tool to classify single cell RNA-Seq data across platforms and across species](https://www.biorxiv.org/content/early/2018/12/31/508085)
* [ONNX](http://onnx.ai/) is a open format to represent deep learning models. With ONNX, AI developers can more easily move models between state-of-the-art tools and choose the combination that is best for them. ONNX is developed and supported by a community of partners.
* [Scalable analysis of cell-type composition from single-cell transcriptomics using deep recurrent learning](https://www.nature.com/articles/s41592-019-0353-7) We present scScope, a scalable deep-learning-based approach that can accurately and rapidly identify cell-type composition from millions of noisy single-cell gene-expression profiles.
* [Deep learning applications in single-cell omics data analysis](https://www.biorxiv.org/content/10.1101/2021.11.26.470166v1) " Through a systematic literature review, we have found that DL has not yet revolutionized or addressed the most pressing challenges of the SC omics field. However, using DL models for single-cell omics has shown promising results (in many cases outperforming the previous state-of-the-art models) but lacking the needed biological interpretability in many cases."
* [Can Large Language Models Reason and Plan?](https://arxiv.org/abs/2403.04121)

### machine learning in life science

* CREME: Cis-Regulatory Element Model Explanations https://github.com/p-koo/creme-nn
* Sensitive tumor detection, accurate quantification, and cancer subtype classification using low-pass whole methylome sequencing of plasma DNA https://www.biorxiv.org/content/10.1101/2024.06.10.598204v1
* [Uncovering expression signatures of synergistic drug responses via ensembles of explainable machine-learning models](https://www.nature.com/articles/s41551-023-01034-0)
* [DeepAIR: A deep learning framework for effective integration of sequence and 3D structure to enable adaptive immune receptor analysis
](https://www.science.org/doi/10.1126/sciadv.abo5128)
* [A book Deep Learning for the Life Sciences: Applying Deep Learning to Genomics, Microscopy, Drug Discovery, and More](https://www.amazon.com/Deep-Learning-Life-Sciences-Microscopy/dp/1492039837)
* [Deep learning shapes single-cell data analysis](https://www.nature.com/articles/s41580-022-00466-x)
* [A call for deep-learning healthcare](https://www.nature.com/articles/s41591-018-0320-3)
* [High-performance medicine: the convergence of human and artificial intelligence](https://www.nature.com/articles/s41591-018-0300-7)
* [Privacy in the age of medical big data](https://www.nature.com/articles/s41591-018-0272-7)
* [Automated identification of Cell Types in Single Cell RNA Sequencing](https://www.biorxiv.org/content/10.1101/532093v1)
* [Deep learning: new computational modelling techniques for genomics](https://www.nature.com/articles/s41576-019-0122-6)
* [immuneML: an ecosystem for machine learning analysis of adaptive immune receptor repertoires](https://www.biorxiv.org/content/10.1101/2021.03.08.433891v1)
* [Machine learning for deciphering cell heterogeneity and gene regulation](https://www.nature.com/articles/s43588-021-00038-7)
* [Gene expression based inference of cancer drug sensitivity](https://www.nature.com/articles/s41467-022-33291-z)
* [Learning the Drug-Target Interaction Lexicon](https://www.biorxiv.org/content/10.1101/2022.12.06.519374v1) accurately scan ~10 million drug-target pairs per minute
* [AI in Drug Discovery 2022 - A Highly Opinionated Literature Review](http://practicalcheminformatics.blogspot.com/2023/01/ai-in-drug-discovery-2022-highly.html)
* [CodonBERT: Large Language Models for mRNA design and optimization](https://www.biorxiv.org/content/10.1101/2023.09.09.556981v1)
* [How to represent a protein sequence](https://liambai.com/protein-representation/)
  
### deep learning for single-cell

* [awesome-deep-learning-single-cell-papers](https://github.com/OmicsML/awesome-deep-learning-single-cell-papers#gan-or-diffusion-model)
* [Predicting RNA-seq coverage from DNA sequence as a unifying model of gene regulation”](https://www.biorxiv.org/content/10.1101/2023.08.30.555582v1) twitter thread to understand https://twitter.com/drklly/status/1698013780394041561
* Single-cell gene expression prediction from DNA sequence at large contexts  https://www.biorxiv.org/content/10.1101/2023.07.26.550634v1.full
* Transfer learning enables predictions in network biology https://pubmed.ncbi.nlm.nih.gov/37258680/
* [Cell2Sentence: Teaching Large Language Models the Language of Biology](https://www.biorxiv.org/content/10.1101/2023.09.11.557287v1)
* [CPA is a framework to learn the effects of perturbations at the single-cell level. CPA encodes and learns phenotypic drug responses across different cell types, doses, and combinations](https://github.com/theislab/cpa)
* [GET: a foundation model of transcription across human cell types](https://www.biorxiv.org/content/10.1101/2023.09.24.559168v1)
* [Neural optimal transport predicts perturbation responses at the single-cell level](https://www.nature.com/articles/s41592-023-01968-y)
* [CellPLM: Pre-training of Cell Language Model Beyond Single Cells](https://www.biorxiv.org/content/10.1101/2023.10.03.560734v1)
* [GEARS: Predicting transcriptional outcomes of novel multi-gene perturbations](https://github.com/snap-stanford/GEARS)
* [Learning single-cell perturbation responses using neural optimal transport](https://www.nature.com/articles/s41592-023-01969-x)
* [Assessing the limits of zero-shot foundation models in single-cell biology](https://www.biorxiv.org/content/10.1101/2023.10.16.561085v1)
* [Evaluating the Utilities of Large Language Models in Single-cell Data Analysis](https://www.biorxiv.org/content/10.1101/2023.09.08.555192v3)
* [GENEPT: A SIMPLE BUT HARD-TO-BEAT FOUNDATION MODEL FOR GENES AND CELLS BUILT FROM CHATGPT](https://www.biorxiv.org/content/10.1101/2023.10.16.562533v1)

  
### Large language models (LMM)  

* [A High-level Overview of Large Language Models](https://www.borealisai.com/research-blogs/a-high-level-overview-of-large-language-models/)
* [Training and fine-tuning large language models](https://www.borealisai.com/research-blogs/training-and-fine-tuning-large-language-models/)
* [A Hackers' Guide to Language Models](https://www.youtube.com/watch?v=jkrNMKz9pWU)
* [SaProt: Protein Language Modeling with Structure-aware Vocabulary](https://www.biorxiv.org/content/10.1101/2023.10.01.560349v2)
* [ProteinBERT: a universal deep-learning model of protein sequence and function](https://academic.oup.com/bioinformatics/article/38/8/2102/6502274)
* [Learning meaningful representations of protein sequences](https://www.nature.com/articles/s41467-022-29443-w)
* [Evaluating the Impact of Sequence Convolutions & Embeddings on Protein Structure Prediction](https://wandb.ai/koes-group/protein-transformer/reports/Evaluating-the-Impact-of-Sequence-Convolutions-Embeddings-on-Protein-Structure-Prediction--Vmlldzo2OTg4Nw) youtube video https://www.youtube.com/watch?v=w0N-N6J05gk
* [Transformer protein language models are unsupervised structure learners](https://www.biorxiv.org/content/10.1101/2020.12.15.422761v1)
* [A 5’ UTR Language Model for Decoding Untranslated Regions of mRNA and Function Predictions](https://www.biorxiv.org/content/10.1101/2023.10.11.561938v1)

### Graph Neural Nets (GNN)

* [A Gentle Introduction to Graph Neural Networks](https://distill.pub/2021/gnn-intro/)
* [Graph Neural Net works](https://mbernste.github.io/posts/gcn/) a blog post by Matt B.
* [DeepCCI: a deep learning framework for identifying cell-cell interactions from single-cell RNA sequencing data](https://academic.oup.com/bioinformatics/advance-article/doi/10.1093/bioinformatics/btad596/7281356?login=false). It uses GNN.
* [Bering: joint cell segmentation and annotation for spatial transcriptomics with transferred graph embeddings](https://www.biorxiv.org/content/10.1101/2023.09.19.558548v1)
* [GOAT: Gene-level biomarker discovery from multi-Omics data using graph ATtention neural network for eosinophilic asthma subtype](https://academic.oup.com/bioinformatics/advance-article/doi/10.1093/bioinformatics/btad582/7280697?login=false)

### datasets for practicing ML

* [Genomic and transcriptomic analysis of checkpoint blockade response in advanced non-small cell lung cancer](https://www.nature.com/articles/s41588-023-01355-5#data-availability) https://zenodo.org/record/7625517#.ZDL9JOzMJsU
